amqp.go

```diff
@@ -1,17 +1,119 @@
 package amqp
 
 import (
+	amqpDriver "github.com/streadway/amqp"
 	"go.k6.io/k6/js/modules"
 )
 
 const version = "v0.0.1"
 
 type Amqp struct {
-    Version string
+	Version    string
+	Connection *amqpDriver.Connection
+	Queues     *Queues
+	Exchanges  *Exchanges
+}
+
+type AmqpOptions struct {
+	ConnectionUrl string
+}
+
+type PublishOptions struct {
+	QueueName string
+	Body      string
+	Exchange  string
+	Mandatory bool
+	Immediate bool
+}
+
+type ConsumeOptions struct {
+	Consumer  string
+	AutoAck   bool
+	Exclusive bool
+	NoLocal   bool
+	NoWait    bool
+	Args      amqpDriver.Table
+}
+
+type ListenerType func(string) error
+
+type ListenOptions struct {
+	Listener  ListenerType
+	QueueName string
+	Consumer  string
+	AutoAck   bool
+	Exclusive bool
+	NoLocal   bool
+	NoWait    bool
+	Args      amqpDriver.Table
+}
+
+func (amqp *Amqp) Start(options AmqpOptions) error {
+	conn, err := amqpDriver.Dial(options.ConnectionUrl)
+	amqp.Connection = conn
+	amqp.Queues.Connection = conn
+	amqp.Exchanges.Connection = conn
+	return err
+}
+
+func (amqp *Amqp) Publish(options PublishOptions) error {
+	ch, err := amqp.Connection.Channel()
+	if err != nil {
+		return err
+	}
+	defer ch.Close()
+
+	return ch.Publish(
+		options.Exchange,
+		options.QueueName,
+		options.Mandatory,
+		options.Immediate,
+		amqpDriver.Publishing{
+			ContentType: "text/plain",
+			Body:        []byte(options.Body),
+		},
+	)
+}
+
+func (amqp *Amqp) Listen(options ListenOptions) error {
+	ch, err := amqp.Connection.Channel()
+	if err != nil {
+		return err
+	}
+	defer ch.Close()
+
+	msgs, err := ch.Consume(
+		options.QueueName,
+		options.Consumer,
+		options.AutoAck,
+		options.Exclusive,
+		options.NoLocal,
+		options.NoWait,
+		options.Args,
+	)
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		for d := range msgs {
+			options.Listener(string(d.Body))
+		}
+	}()
+	return nil
 }
 
 func init() {
-	modules.Register("k6/x/amqp", &Amqp{
-		Version: version,
-	})
+
+	queues := Queues{}
+	exchanges := Exchanges{}
+	generalAmqp := Amqp{
+		Version:   version,
+		Queues:    &queues,
+		Exchanges: &exchanges,
+	}
+
+	modules.Register("k6/x/amqp", &generalAmqp)
+	modules.Register("k6/x/amqp/queues", &queues)
+	modules.Register("k6/x/amqp/exchanges", &exchanges)
```

Comment:
```suggestion
	modules.Register("k6/x/amqp/queue", &queues)
	modules.Register("k6/x/amqp/exchange", &exchanges)
```

Use singularis throughout unless the pluralization adds semantic value.